src/web/components/Core/ActionButton.tsx

```diff
@@ -0,0 +1,31 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import clsx from 'clsx';
+
+import '../../styles/buttons.scss';
+
+export enum ActionButtonIcon {
+  edit = 'pencil',
+  delete = 'trash-can'
+};
+
+type ActionButtonProps = Readonly<React.ButtonHTMLAttributes<HTMLButtonElement>> & {
+  icon: ActionButtonIcon;
```

Comment:
I'm not sure if we need this `ActionButtonIcon` enum - I think it is probably better to keep it more generic but also preserve type safety just have the `icon` prop be the type `IconProp`, like we have in `Notification.tsx`. Because that's the type that ends up getting passed into the `<FontAwesomeIcon>` anyway.

So I think this is better:
```
type ActionButtonProps = Readonly<React.ButtonHTMLAttributes<HTMLButtonElement>> & {
  icon: IconProp;
  iconClassName?: string;
};
```

Then the usage would remain something like:
```
<ActionButton
  onClick={onOpenChangeKeyEditDialog}
  icon='pencil'
  className='action-button'
/>
```

Reply:
Makes sense - thanks!